lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -689,6 +685,16 @@ def find_branch(search_list)
 
     class Tree
 
+      def self.tree_entry_from_rugged_hash(entry, root = '')
```

Comment:
I'm not loving this name of this class method. One reason is that the name suggests that the method returns a tree entry based on hash input ("from hash"), but the name of the first argument and the body of the method imply that an `entry` (whatever that is) is transformed into a hash (thus the other way around).

A second reason is that I have no sense of what a `rugged_hash` is. It seems as if is is an ordinary hash originating from a method in `rugged`, but then I don't see why we need to know the origin of the hash.

Would it make sense to rename this to something like `tree_entry_to_hash(entry, root='')`?

Reply:
Good points, thanks! The wording is tricky since a tree entry is essentially just a hash representation of an object in a git tree, as used by rugged, and we want to convert this into our own hash representation of an object in a git tree. `tree_entry_to_hash` misleadingly sounds as if what we're passing in is a hash *instead of* a tree entry.  What about something like `canonicalize_tree_entry`?